backend/grant/commands.py

```diff
@@ -137,3 +138,46 @@ def urls(url, order):
 
     for row in rows:
         click.echo(str_template.format(*row[:column_length]))
+
+
+@click.command()
+@with_appcontext
+def reset_db_chain_data():
+    """Removes chain-state dependent entities from the database. Cannot be undone!"""
+    from grant.extensions import db
+    from grant.proposal.models import Proposal
+    from grant.user.models import UserSettings
+    from grant.task.models import Task
+
+    # Delete all proposals. Should cascade to contributions, comments etc.
+    p_count = 0
+    for proposal in Proposal.query.all():
+        db.session.delete(proposal)
+        p_count = p_count + 1
+
+    # Delete all outstanding tasks
+    t_count = Task.query.delete()
```

Comment:
Didn't know `delete` gave you the count!

Reply:
Indeed it does. I would have done the same for proposals, but in order to get cascade behavior to work correctly, you have to do `db.session.delete`, not sql query delete.

Reply:
I see, good to know